### ISSUE TITLE
What are Light Zone minimum and optimal PC configuration ?

### ISSUE THREAD

**Issue by Maqua66** [ktgw0316/LightZone#304]
Hello my PC died some time ago and I used it to store and lightly edit some photos. I use Light Zone so I would like to know the minimum and optimal PC configuration for this logiciel to buy a PC wich will handle better the tasks I give it than my former one.

**Comment by jedd** [comment]
Hi @Maqua66 - I don't believe we'd have anything too prescriptive to advise here, as performance would depend on the nature of your files (sizes), complexity of your stacks, tolerance to waiting for the computer etc.

Assuming your budget today exceeds the like-for-like replacement cost of your old computer, you probably can't go wrong.  I'd suggest additional memory and a fast (solid state) hard disk probably influence performance more strongly than CPU.

Some datapoints of dubious value -- I move between a 10yo Xeon E3-1245 (32GB) desktop, a 3yo Ryzen 5600x (32GB) desktop, and a 10yo i7 (16GB) laptop, all with SSD drives.  _For my use cases_ of LightZone they all feel plenty performant.

**Comment by sfink16** [comment]
> I'd suggest additional memory and a fast (solid state) hard disk probably influence performance more strongly than CPU.

This is interesting @jedd.  I don't want to buy a new PC however.  I'm at the max of 8 GB of memory for my low end desktop.  Can an external SSD work?  I'm guessing that I can install LZ on an external SSD.

Steve

**Comment by jedd** [comment]
External SSD via USB? You'll be limited by the speed of the USB connection, and if your machine maxes out at 8GB memory I'm assuming it doesn't have USB3.  Easy enough to try, but I wouldn't expect miracles from that arrangement - much better to have an SSD attached to a faster bus (nvme, sata) - are you sure there isn't capacity inside the machine for an internal upgrade?   Typically moving to SSD is one of the cheaper upgrades you can do to do get a boost in performance from an older computer.

Again, as per OP, actual real world performance is going to come back to complexity of your stack, appetite for waiting for the computer to respond, size of your images, etc.

I'll note we're getting a touch off-topic from LightZone here - @Maqua66 are you happy to close this ticket?

**Comment by sfink16** [comment]
> External SSD via USB? You'll be limited by the speed of the USB connection, and if your machine maxes out at 8GB memory I'm assuming it doesn't have USB3. Easy enough to try, but I wouldn't expect miracles from that arrangement - much better to have an SSD attached to a faster bus (nvme, sata) - are you sure there isn't capacity inside the machine for an internal upgrade? Typically moving to SSD is one of the cheaper upgrades you can do to do get a boost in performance from an older computer.

Below is the link for my Acer Aspire M3470G desktop.  Yes, it does have 2 USB 3.0 slots available.  It also list the Sata interface as "SATA X 3(SATA 6Gb/s)", so I'm assuming that means 3 drives can be added.  [Here is the link](https://www.manualslib.com/manual/370374/Acer-Aspire-M3470.html?page=16&term=Usb+Port&selected=1#manual)

I also was surprised that this "discussion" was not added to the discussion tab instead of this "Issues" tab.

Steve

**Comment by jedd** [comment]
Hey @sfink16 - yes, USB3 may work fine - I've had bad experiences with vendors of both motherboards and USB sticks / drives over-stating performance of USB.  In this situation, unless you need the portability benefits of an external drive, an internal drive should be slightly cheaper (you're not paying for the enclosure) and more performant (a delta of 4.8GBps vs 6GBps - very roughly - there are LOTS of discussions about the performance of these things on the net).

Even more broadly than 'about LightZone performance' - replacing the existing spinning rust internal drive with an SSD would obviously improve performance of the OS & other apps too.  There are tools that ship with Debian GNU/Linux that let you work out where your performance is being throttled (cpu, memory, i/o, etc).

**Comment by sfink16** [comment]
@jedd - I accidentally wiped out my Windows installation when I installed Ubuntu trying to do a dual boot, so I'm familiar with Linux tools (HTop, System Monitor, etc.).  I do have a HP Mini (Windows 10) sitting a couple of feet away, so portability is preferable in my situation.

**Comment by tbpassin** [comment]
USB SSD drives are usually much slower than internal SATA SSDs.  But one would most likely be better than an old magnetic disk drive.  Also a SSD on USB is better (faster) than a USB memory stick, at least the ones I've tried.

I have a 12-year old Sony Vaio that I've been running off an external SSD.  I mounted Linux Mint on it.  It is much snappier than it used to be with Windows 8 and an internal magnetic disk.  Of course, for memory- or CPU-bound tasks the drive isn't going to help much ...
